packages/engine/src/scene/systems/ShadowSystem.tsx

```diff
@@ -451,6 +434,41 @@ const execute = () => {
     return
   }
 
+  for (const entity of directionalLightQuery.enter()) {
```

Comment:
Using enter and exit queries is a practice we're trying to avoid for the exact reason the comment here specifies. I think this can be simplified in the studio by having a reactor run that listens for the entity's existence specified by the `primaryLight` field on the RenderSettingsComponent, setting it to `""` if the entity is removed, and if the value is `""` and then if the user adds a `DirectionalLightComponent`, it auto populates that field.